Add a guide to GKE Auto-Discovery

ptgott opened the pull request:
Closes #18011

ptgott committed:
Add a guide to GKE Auto-Discovery

Closes #18011

tigrato commented on docs/pages/kubernetes-access/discovery/google-cloud.mdx:
```diff
@@ -0,0 +1,464 @@
+title: GKE Cluster Access Manager
+description: "Manage access to GKE clusters"
+stage: GA
+includedPermissions:
```
Kubernetes Service also needs `container.clusters.get`.

The AKS and EKS guides show this step as unique permissions for both services. Keeping everything uniform we should show the permissions mixed here or separated in the other two cases to maintain consistency.

tigrato commented on docs/pages/kubernetes-access/discovery/google-cloud.mdx:
```diff
@@ -0,0 +1,464 @@
+
+### Download a credentials file
+
+Download a credentials file for the service account used by the Discovery
```
Make a notice block informing that this is not required if the goal is to run Teleport services in a GCP VM

ptgott replied:
Would you still need to attach the service account to the GCP VM? I've added these instructions to attach the service account: https://cloud.google.com/compute/docs/access/create-enable-service-accounts-for-instances#changeserviceaccountandscopes

tigrato replied:
Yes

tigrato commented on docs/pages/kubernetes-access/discovery/google-cloud.mdx:
````diff
@@ -0,0 +1,464 @@
+```yaml
+discovery_service:
+  enabled: "yes"
+  gcp:
````
It will be good to extend AWS and Azure guides with the same detail as GCP

ptgott replied:
I can create an issue for this. Are there any particular areas we should concentrate on when expanding the AWS and Azure guides?

ptgott replied:
I'll add to this issue to include separate roles for the Discovery/Kubernetes Service in the AWS/Azure guides per https://github.com/gravitational/teleport/pull/18582#discussion_r1029443620. I think separate roles are appropriate in this case to clarify that these services do different things and make it easier to run the services separately.

tigrato replied:
I agree with you.

ptgott commented:
**TODO:**
- [ ] Test this guide with a GCP VM and add separate `TabItem`s for GCP instructions (namely, attaching the service accounts to a GCP VM instead of using a credentials file)
- [ ] Where relevant, extract content shared between this guide and my Standalone Kubernetes guide changes into partials (https://github.com/gravitational/teleport/pull/18608).
- [ ] Respond to @tigrato's feedback

ptgott committed:
Respond to PR feedback

ptgott committed:
Update the guide while testing it

- Add the `scopes` flag to the `gclodu compute instances
  set-service-account command`
- Replace `auth_servers` with `proxy_server` in the config

alexfornuto commented on docs/pages/kubernetes-access/discovery/google-cloud.mdx:
````diff
@@ -0,0 +1,578 @@
+Stop your VM so you can attach your service account to it:
+
+```code
+$ VM_NAME=my-vm
+$ MY_ZONE=my-region
````
Can we use the Var component here?

alexfornuto replied:
I see you are now using the Var component to set the shell var, but why not just use the Var directly in the command? What's the value of setting it in the shell to be used just once immediately after?

ptgott replied:
I had thought it would get pretty cluttered to include two `Var`s on one line, but it turns out that it display just fine 👍

alexfornuto commented on docs/pages/kubernetes-access/discovery/google-cloud.mdx:
```diff
@@ -0,0 +1,578 @@
+  --description="Teleport Discovery Service" \
+  --display-name="teleport-discovery-service"
+$ gcloud iam service-accounts create teleport-kubernetes-service \
+  --description="Teleport Discovery Service" \
+  --display-name="teleport-discovery-service"
```
```suggestion
  --description="Teleport Kubernetes Service" \
  --display-name="teleport-kubernetes-service"
```

ptgott replied:
Oof, good catch

alexfornuto commented on docs/pages/kubernetes-access/discovery/google-cloud.mdx:
````diff
@@ -0,0 +1,578 @@
+Stop each VM you plan to use to run the Teleport Kubernetes Service and
+Discovery Service.
+
+Attach your service account to the VM running the Kubernetes Service:
+
+```code
+$ gcloud compute instances set-service-account ${VM1_NAME?} \
+   --service-account teleport-kubernetes-service@${PROJECT_ID?}.iam.gserviceaccount.com \
+   --zone ${MY_ZONE?} \
+   --scopes=cloud-platform
+```
+
+Attach your service account to the VM running the Discovery Service:
````
```suggestion
Attach your "Teleport Discovery Service" service account to the VM running the Kubernetes Service:

```code
$ gcloud compute instances set-service-account ${VM1_NAME?} \
   --service-account teleport-kubernetes-service@${PROJECT_ID?}.iam.gserviceaccount.com \
   --zone ${MY_ZONE?} \
   --scopes=cloud-platform
```

Attach your "Teleport Kubernetes Service" service account to the VM running the Discovery Service:
```

alexfornuto commented on docs/pages/kubernetes-access/discovery/google-cloud.mdx:
````diff
@@ -0,0 +1,578 @@
+`tctl` command:
+
+```code
+$ tctl tokens add --type=discovery,kube
````
For those running Discovery and Kubernetes services separately, does this command/section need another version in a Details block, split out into tokens for each service

alexfornuto commented on docs/pages/kubernetes-access/discovery/google-cloud.mdx:
```diff
@@ -0,0 +1,578 @@
+
+### Configure the Kubernetes Service and Discovery Service
+
+On the host running the Kubernetes Service and Discovery Service, create a
```
As above, does this need a separate services version?

alexfornuto commented on docs/pages/kubernetes-access/discovery/google-cloud.mdx:
```diff
@@ -0,0 +1,578 @@
+
+</Details>
+
+(!docs/pages/includes/kubernetes-access/rbac.mdx!)
```
It's not part of this PR but I noticed it while reviewing this doc... This Partial uses a Var component to set a shell Var to then use in a single command:

````
```code
# e.g., myuser
$ TELEPORT_USER=<Var name="teleport-user"/>
$ tctl get user/${TELEPORT_USER?} > user.yaml
```
````

Why not just have the command use the Var component value?

````
```code
# e.g., myuser
$ tctl get user/<Var name="teleport-user"/> > user.yaml
```
````

ptgott committed:
Respond to PR feedback

alexfornuto reviewed (approved):
👍 , less the outstanding discussion on the value of making single-use shell vars.

ptgott committed:
Add Vars and remove "Preview"

ptgott commented:
- [ ] TODO: Mark as preview when I backport to v11
